File: ClientApp/src/app/interfaces/product/userInteract/response/orderResponse.ts
import { AddressResponse } from "../../../Security/Response/addressResponse";
import { BaseResponse } from "../../../base-response";
import { PurchaseDetailResponse } from "./purchaseDetailResponse";

export interface OrderResponse extends BaseResponse {
    sKU: number;
    payMethodId: string;
    userId: string;
    orderDate: string;
    shippingCost: number;
    voucherType: string;
    voucherNumber: number;
    voucherSeries: string;
    ammountPaid: number;
    taxesCost: number;
    details: PurchaseDetailResponse[];
    addressId: string;
    address: AddressResponse;
}

File: ClientApp/src/app/interfaces/Product/UserInteract/Response/OrderResponse.cs
using Contracts.Model.Security.Response;

namespace Contracts.Model.Product.UserInteract.Response;

public class OrderResponse : BaseResponse
{
	public Guid Id { get; set; }
	public Guid PayMethodId { get; set; }
	public int State { get; set; }
	public string UserId { get; set; }
	public double ShippingCost { get; set; }
	public string VoucherType { get; set; }
	public int VoucherNumber { get; set; }
	public string VoucherSeries { get; set; }
	public double TaxesCost { get; set; }
	public ICollection<PurchaseDetailResponse> Details { get; set; }
	public Guid AddressId { get; set; }
	public AddressResponse Address { get; set; }
}